src/test/mir-opt/simplify-locals-removes-unused-discriminant-reads.rs

```diff
@@ -0,0 +1,65 @@
+fn map(x: Option<Box<()>>) -> Option<Box<()>> {
+    match x {
+        None => None,
+        Some(x) => Some(x),
+    }
+}
+
+fn main() {
+    map(None);
+}
+
+// END RUST SOURCE
+// START rustc.map.SimplifyLocals.before.mir
+// debug x => _1;
+// let mut _0: std::option::Option<std::boxed::Box<()>>;
+// let mut _2: isize;
+// let _3: std::boxed::Box<()>;
+// let mut _4: std::boxed::Box<()>;
+// let mut _5: isize;
+// let mut _6: isize;
+// scope 1 {
+//   debug x => _3;
+// }
+// bb0: {
+//   _2 = discriminant(_1);
+//   switchInt(move _2) -> [0isize: bb2, otherwise: bb1];
+// }
+// bb1: {
+//   _0 = move _1;
+//   goto -> bb3;
+// }
+// bb2: {
+//   discriminant(_0) = 0;
+//   goto -> bb3;
+// }
+// bb3: {
+//   _5 = discriminant(_1);
+//   return;
+// }
+// END rustc.map.SimplifyLocals.before.mir
+
+// START rustc.map.SimplifyLocals.after.mir
+// debug x => _1;
+// let mut _0: std::option::Option<std::boxed::Box<()>>;
+// let mut _2: isize;
+// let _3: std::boxed::Box<()>;
+// scope 1 {
+//   debug x => _3;
+// }
+// bb0: {
+//   _2 = discriminant(_1);
+//   switchInt(move _2) -> [0isize: bb2, otherwise: bb1];
+// }
+// bb1: {
+//   _0 = move _1;
+//   goto -> bb3;
+// }
+// bb2: {
+//   discriminant(_0) = 0;
+//   goto -> bb3;
+// }
+// bb3: {
+//   return;
+// }
+// END rustc.map.SimplifyLocals.after.mir
```

Comment:
```suggestion
// EMIT_MIR rustc.map.SimplifyLocals.diff
```

And run with `--bless`

Reply:
That is beautiful! ❤️